### ISSUE TITLE
 .bss segment place psram, flash erase errors (IDFGH-3878)

### ISSUE THREAD

**Issue by DURBIN-LI** [espressif/esp-idf#5779]
## Environment

- Development Kit:      [none] **ESP32-S AI-Thinker**
- Kit version (for WroverKit/PicoKit/DevKitC): [nonr]
- Module or chip used:  [ESP32-S]
- IDF version : 		v4.2-dev-1575-ga3520970f
- Build System:         [idf.py]
- Compiler version :xtensa-esp32-elf-gcc (crosstool-NG esp-2020r1) 8.2.0
- Operating System:     [Linux]
- Using an IDE?: [No]
- Power Supply:         [USB|external 5V|external 3.3V|Battery]

## Problem Description

- enable allow .bss segment place in external memory, OTA upgradeing, flash erase will appear program errors,system reset

### Expected Behavior

- enable allow .bss segment place in external memory,  OTA upgraded

### Actual Behavior

- tracking  program, function`esp_ota_begin`call interface `esp_partition_erase_range`; then system reset

### Code to reproduce this issue
#include "esp_ota_ops.h"
```cpp

esp_err_t esp_ota_begin(const esp_partition_t *partition, size_t image_size, esp_ota_handle_t *out_handle)
{
    ota_ops_entry_t *new_entry;
    esp_err_t ret = ESP_OK;

ESP_LOGI(TAG, "1");
    if ((partition == NULL) || (out_handle == NULL)) {
        return ESP_ERR_INVALID_ARG;
    }

    partition = esp_partition_verify(partition);
    if (partition == NULL) {
        return ESP_ERR_NOT_FOUND;
    }

    if (!is_ota_partition(partition)) {
        return ESP_ERR_INVALID_ARG;
    }

ESP_LOGI(TAG, "2");

    const esp_partition_t* running_partition = esp_ota_get_running_partition();
    if (partition == running_partition) {
        return ESP_ERR_OTA_PARTITION_CONFLICT;
    }

#ifdef CONFIG_BOOTLOADER_APP_ROLLBACK_ENABLE
    esp_ota_img_states_t ota_state_running_part;
    if (esp_ota_get_state_partition(running_partition, &ota_state_running_part) == ESP_OK) {
        if (ota_state_running_part == ESP_OTA_IMG_PENDING_VERIFY) {
            ESP_LOGE(TAG, "Running app has not confirmed state (ESP_OTA_IMG_PENDING_VERIFY)");
            return ESP_ERR_OTA_ROLLBACK_INVALID_STATE;
        }
    }
#endif

#if 1
ESP_LOGI(TAG, "3");

    // If input image size is 0 or OTA_SIZE_UNKNOWN, erase entire partition
    if ((image_size == 0) || (image_size == OTA_SIZE_UNKNOWN)) {
ESP_LOGI(TAG, "3.1 = %X %d",partition->address,partition->size);
        ret = esp_partition_erase_range(partition, 0, partition->size);
    } else {
        const int aligned_erase_size = (image_size + SPI_FLASH_SEC_SIZE - 1) & ~(SPI_FLASH_SEC_SIZE - 1);
        ret = esp_partition_erase_range(partition, 0, aligned_erase_size);
    }
ESP_LOGI(TAG, "4 = %X",ret);

    if (ret != ESP_OK) {
        return ret;
    }
#endif


    new_entry = (ota_ops_entry_t *) calloc(sizeof(ota_ops_entry_t), 1);
    if (new_entry == NULL) {
        return ESP_ERR_NO_MEM;
    }

    LIST_INSERT_HEAD(&s_ota_ops_entries_head, new_entry, entries);

    if ((image_size == 0) || (image_size == OTA_SIZE_UNKNOWN)) {
        new_entry->erased_size = partition->size;
    } else {
        new_entry->erased_size = image_size;
    }
ESP_LOGI(TAG, "5");
    new_entry->part = partition;
    new_entry->handle = ++s_ota_ops_last_handle;
    *out_handle = new_entry->handle;
    return ESP_OK;
}

```

## Debug Logs

**ERR LOG**
```
I (165) cam_ota: Running partition type 0 subtype 16 (offset 0x00010000)

[10:38:30.569]I (202) cam_ota: heap size = 4306728

[10:38:30.612]I (213) cam_ota: Writing to partition subtype 17 at offset 0x110000
I (238) cam_ota: New firmware version: beta_v1.0.1-105-g1fc19e7-dirty
I (243) cam_ota: Running firmware version: beta_v1.0.1-108-gcf6c492-dirty
I (275) cam_ota: ota begin
I (128) esp_ota_ops: 1
I (142) esp_ota_ops: 2
I (160) esp_ota_ops: 3
I (164) esp_ota_ops: 3.1 = 110000 1048576
I (426) partition: erase region
E (319) spi_flash: >Check for write protected regions

[10:38:31.571]ets Jun  8 2016 00:22:57

rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
```

**POWER ON LOG**
```
rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4
load:0x3fff0034,len:6600
load:0x40078000,len:14524
load:0x40080400,len:4628
entry 0x400806a4
I (100) boot: ESP-IDF v4.2-dev-1575-ga3520970f-dirty 2nd stage bootloader
I (101) boot: compile time 18:02:58
I (52) boot: chip revision: 1
I (297) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (210) qio_mode: Enabling default flash chip QIO
I (208) boot.esp32: SPI Speed      : 80MHz
I (226) boot.esp32: SPI Mode       : QIO
I (248) boot.esp32: SPI Flash Size : 4MB
I (94) boot: Enabling RNG early entropy source...
I (135) boot: Partition Table:
I (136) boot: ## Label            Usage          Type ST Offset   Length
I (201) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (201) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (201) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (201) boot:  3 ota_0            OTA app          00 10 00010000 00100000
I (201) boot:  4 ota_1            OTA app          00 11 00110000 00100000
I (201) boot:  5 custom_data      unknown          40 00 00210000 001f0000
I (206) boot: End of partition table
I (297) boot_comm: chip revision: 1, min. application chip revision: 0
I (520) esp_image: segment 0: paddr=0x00010020 vaddr=0x3f400020 size=0x26664 (157284) map

[10:17:15.698]I (520) esp_image: segment 1: paddr=0x0003668c vaddr=0x3ffb0000 size=0x044d0 ( 17616) load
I (520) esp_image: segment 2: paddr=0x0003ab64 vaddr=0x40080000 size=0x00404 (  1028) load
I (520) esp_image: segment 3: paddr=0x0003af70 vaddr=0x40080404 size=0x050a8 ( 20648) load
I (520) esp_image: segment 4: paddr=0x00040020 vaddr=0x400d0020 size=0xa91d4 (692692) map

[10:17:15.949]I (520) esp_image: segment 5: paddr=0x000e91fc vaddr=0x400854ac size=0x1964c (104012) load

[10:17:16.006]I (408) boot: Loaded app from partition at offset 0x10000
I (625) boot: Disabling RNG early entropy source...
I (827) psram: This chip is ESP32-D0WD
I (165) spiram: Found 64MBit SPI RAM device
I (168) spiram: SPI RAM mode: flash 80m sram 40m
I (172) spiram: PSRAM initialized, cache is in low/high (2-core) mode.

```
tks，best wish

**Comment by Alvin1Zhang** [comment]
@DURBIN-LI Thanks for reporting. Would you please help provide more details as suggested in the [issue template](https://github.com/espressif/esp-idf/issues/new?assignees=&labels=&template=bug_report.md&title=)? Information like elf, sdk configuration, backtrace, log outputs, commit ID, hardware and etc. would help us debug further. Thanks.

**Comment by DURBIN-LI** [comment]
@Alvin1Zhang update issue

**Comment by DURBIN-LI** [comment]
@Alvin1Zhang issue has been update

**Comment by Alvin1Zhang** [comment]
Thanks for updates and providing more details, we will look into.

**Comment by shubhamkulkarni97** [comment]
Hi @DURBIN-LI,

I could reproduce the issue by checking out to the commit you mentioned (i.e. `v4.2-dev-1575-ga3520970f`). However, this issue is fixed and is not observed on latest release/v4.2 branch.

Updating your branch to latest commit should fix the issue.

Thanks,
Shubham

**Comment by DURBIN-LI** [comment]
@shubhamkulkarni97 Hi，Updated v4.2 branch，the issue has been fixed
tks，best wish
